### PR TITLE
Clarified sampling process of torch.randn for complex dtypes.

### LINKED ISSUE #118269: Incomplete documentation for `torch.randn` on complex dtypes
### 📚 The doc issue

https://pytorch.org/docs/stable/generated/torch.randn.html does not clearly specify behavior on complex data types `torch.complex64` and `torch.complex128`. 

The current notation $\textrm{out}_{i} \sim \mathcal{N} (0, 1)$ may lead to users thinking that both the real and imaginary parts are sampled from a unit Gaussian, but in fact they are each i.i.d. sampled from $\mathcal{N} (0, 1/2)$, leading to the complex signal being (correctly, for a large class of applications) sampled from the complex random normal distribution.

The following basic snippet validates this:

```
import torch

x = torch.randn(1000, 1000, dtype=torch.complex64)
print('Complex signal power using native complex: ', torch.var(x))
print('Real part signal power using native complex: ', torch.var(torch.real(x)))
print('Imag part signal power using native complex: ', torch.var(torch.imag(x)))
```

It will produce:
```
Complex signal power using native complex:  tensor(0.9998)
Real part signal power using native complex:  tensor(0.5005)
Imag part signal power using native complex:  tensor(0.4993)
```

The same holds for `torch.complex128`.

### Suggest a potential alternative/fix

The documentation for `torch.randn` could be updated to include something similar to the following clarifications:

- For complex `dtype`, the tensor is i.i.d. sampled from a complex normal distribution with zero mean and unit variance as: $\textrm{out}_{i} \sim \mathcal{CN} (0, 1) $

- $\mathcal{CN}$ is the standard notation for the complex normal distribution (https://en.wikipedia.org/wiki/Complex_normal_distribution).

cc @svekars @brycebortree @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

### PR DESCRIPTION
**Pull request by tringwald**
Fixes #118269.

Clarified the docs of `torch.randn` and `torch.randn_like`.

cc @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -9216,6 +9216,13 @@ def merge_dicts(*dicts):
 
 The shape of the tensor is defined by the variable argument :attr:`size`.
 
+.. note::
```
Comment (lezcano) [inline torch/_torch_docs.py:9219]: No need for this to be a note. It can be part of the main docs. You can also move the paragraph right after the definition of the real part.

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -9216,6 +9216,13 @@ def merge_dicts(*dicts):
...
+    For complex dtypes, the tensor is i.i.d. sampled from a `complex normal distribution`_ with zero mean and
+    unit variance as :math:`\text{{out}}_{{i}} \sim \mathcal{{CN}}(0, 1)`. This is equivalent to
+    :math:`\mathcal{{R}}(\text{{out}}_{{i}}) \sim \mathcal{{N}}(0, \frac{{1}}{{2}})` and
```
Comment (lezcano) [inline torch/_torch_docs.py:9222]: Mind writing the real part as `\operatorname{Re}` and the imaginary part as `\operatorname{Im}`?

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -9216,6 +9216,13 @@ def merge_dicts(*dicts):
 
 The shape of the tensor is defined by the variable argument :attr:`size`.
 
+.. note::
+    For complex dtypes, the tensor is i.i.d. sampled from a `complex normal distribution`_ with zero mean and
+    unit variance as :math:`\text{{out}}_{{i}} \sim \mathcal{{CN}}(0, 1)`. This is equivalent to
```
Comment (lezcano) [inline torch/_torch_docs.py:9221]: Perhaps we could have the defining equations in their own line, similar to what we do in the real case.

**Review by lezcano — COMMENTED** [review]
Thank you! I'm leaving mostly a few stylistic points.

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -9214,8 +9214,22 @@ def merge_dicts(*dicts):
...
+unit variance as
+
+.. math::
+    \text{{out}}_{{i}} \sim \mathcal{{CN}}(0, 1)
+
+This is equivalent to separately sampling the real :math:`(\operatorname{{Re}})` and imaginary
+:math:`(\operatorname{{Im}})` part of :math:`\text{{out}}_i` as
+
+.. math::
+    \operatorname{{Re}}(\text{{out}}_{{i}}) \sim \mathcal{{N}}(0, \frac{{1}}{{2}}),\quad
```
Comment (lezcano) [inline torch/_torch_docs.py:9227]:
```suggestion
    \operatorname{{Re}}(\text{{out}}_{{i}}) \sim \mathcal{{N}}\Big(0, \frac{{1}}{{\sqrt{{2}}}}\Big),\quad
```
Same below

**Review by lezcano — COMMENTED** [review]
Now that I see it with the new format, I realised that the formula was wrong. Note that the complex normal is just the normal distribution on 2D, so we need to normalise the norm of its standard variance.

I also fixed the size of the parens now that we are at it.

File: torch/_torch_docs.py (hunk 4 above)
Comment (tringwald) [reply torch/_torch_docs.py:9227]: I can fix the parens, but the formula should be correct according to [Wikipedia](https://en.wikipedia.org/wiki/Complex_normal_distribution). We also list the other equations as `N(mean, variance)`, so `N(mean, std)` would probably be confusing here.

Comment (lezcano) [reply torch/_torch_docs.py:9227]: Ah, right, we are showing the variance. Yep, then it's right.

**Review by lezcano — APPROVED** [review]

**Comment by lezcano** [comment]
Thank you for the contribution!
